test/e2e/apps/daemon_set.go

```diff
@@ -1070,7 +1074,7 @@ func checkDaemonPodsImageAndAvailability(c clientset.Interface, ds *appsv1.Daemo
 		// Make sure every daemon pod on the node has been updated
 		nodeNames := schedulableNodes(c, ds)
 		for _, node := range nodeNames {
-			if nodesToUpdatedPodCount[node] == 0 {
+			if nodesToUpdatedPodCount[node] != 1 {
```

Comment:
Don't change that bit, I'd only leave the `DeletionTimestamp` checks you've added in those 2 places. This wasn't previously the issue, this was introduced with MaxSurge for rolling update in https://github.com/kubernetes/kubernetes/pull/96441

Reply:
I see, even though this doesn't hurt. I have removed it.